packages/botonic-react/src/webchat-app.jsx

```diff
@@ -185,7 +196,32 @@ export class WebchatApp {
     )
   }
 
-  render(dest, optionsAtRuntime = {}) {
-    render(this.getComponent(optionsAtRuntime), dest)
+  async isWebchatVisible({ appId }) {
+    try {
+      const { status } = await HubtypeService.getWebchatVisibility({
+        appId,
+      })
+      if (status === 200) return true
+      return false
+    } catch (e) {
+      return false
+    }
+  }
+
+  async resolveWebchatVisibility(optionsAtRuntime) {
+    let { appId, visibility } = optionsAtRuntime
+    visibility = visibility || this.visibility
+    if (visibility === undefined) {
+      if (await this.isWebchatVisible({ appId })) return true
+    }
+    if (visibility === true) return true
+    if (typeof visibility === 'function' && visibility()) return true
+    return false
```

Comment:
Take another look at this boolean conditions. Most of them can be simplified. Also, remember that you can always do `Boolean(value)`. Example: `Boolean(undefined)` is `false`

Reply:
I don't see how this can be simplified without affecting readability. `visibility` here can take several values:
* `undefined`: do the default behaviour (check visibility from Hubtype API). It has to be exactly `undefined`, not `false` or other falsy values.
* A function: in this case we need to call that function and set visibility according to the boolean return value of that.
* `true`: visibility on
* "anythinbg else": visibility off